created custom component and hosted json data files

timisenco2015 opened the pull request:
Created
    - FieldsAddress
    - Country
    - City
    - Province
 Created and hosted using json-server
     -  AB.json
     - BC.json
     - citiesandtowns
     - MB.json
     - NB.json
     - NF.json
     - NS.json
     - NT.json
     - NU.json
     - ON.json
     - PE.json
     - provinces.json
     - QC.json
     - SK.json
     - YT.json
    To have select options populated for all the components (FieldsAddress, City, Country, and Province), you will need to run the JSON server;
    - open the command prompt on windows
    - npm install JSON server 
    - cd to common-hosted-form-service\app\countrydataserver
    - enter node server.js
    - must be running on port 3000

<!-- Provide a general summary of your changes in the Title above -->
# Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

## Types of changes

<!-- What types of changes does your code introduce? Uncomment all that apply: -->

<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Documentation (non-breaking change with enhancements to documentation) -->
<!-- Breaking change (fix or feature that would cause existing functionality to change) -->

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

jujaga reviewed (changes requested):
We cannot be introducing side-car programs to host static data as that is a complete anti-pattern to containerized application design; not to mention that side-car program would never be invoked in the container to begin with from a quick glance.

As discussed previously, while form.io custom components are definitely scalable and customizeable, the way they are currently managed and implemented in CHEFS does not allow this to be permutable per instance yet as it must be defined, imported and built during compile time.

Unfortunately I'm going to have to block this PR in the interim; aside from the above architectural issues, there are too many impactful structural changes, "hello world" artifact code, TODOs and unnecessary console.log statements that can potentially break existing form functionality for existing clients.

timisenco2015 commented:
> We cannot be introducing side-car programs to host static data as that is a complete anti-pattern to containerized application design; not to mention that side-car program would never be invoked in the container to begin with from a quick glance.
> 
> As discussed previously, while form.io custom components are definitely scalable and customizeable, the way they are currently managed and implemented in CHEFS does not allow this to be permutable per instance yet as it must be defined, imported and built during compile time.
> 
> Unfortunately I'm going to have to block this PR in the interim; aside from the above architectural issues, there are too many impactful structural changes, "hello world" artifact code, TODOs and unnecessary console.log statements that can potentially break existing form functionality for existing clients.

I have removed the side-car. I thought that the best option was to have the static data in the folder with the source. I actually thought of adding json-server to docker.

I will forward this feedback to my team for review.

The "Hello world" artifact code, TODOs and console.log are from my source code.

jujaga commented:
Closing PR for now - can revisit these concepts down the line.
